src/connection.rs

```diff
@@ -160,12 +160,25 @@ fn listen_on_uni_streams(connection: quinn::Connection, tx: Sender<IncomingMsg>)
             let tx = tx.clone();
 
             // Make sure we are able to process multiple streams in parallel.
-            let _ = tokio::spawn(async move {
-                let msg = WireMsg::read_from_stream(&mut recv).await;
+            let _handle = tokio::spawn(async move {
+                let reserved_sender = match tx.reserve().await {
+                    Ok(p) => p,
+                    Err(error) => {
+                        tracing::error!(
+                            "Could not reserve sender for new conn msg read: {error:?}"
+                        );
+                        return;
+                    }
+                };
+
+                let msg = WireMsg::read_from_stream(recv).await;
 
                 // Send away the msg or error
-                let _ = tx.send(msg.map(|r| (r, None))).await;
+                reserved_sender.send(msg.map(|r| (r, None)));
             });
+
+            // now that work is off thread, and we loop, a tiny pause for CPU sanity
+            tokio::time::sleep(Duration::from_millis(50)).await;
```

Comment:
I don't think we need this, or we shouldn't, if buffer is full then it'll be awaiting when attempting to reserve a lot, otherwise there shouldn't be anything needing such a thing